src/js/timer.js

```diff
@@ -0,0 +1,64 @@
+/*******************************************************************************
+
+    uBlock Origin - a browser extension to block requests.
+    Copyright (C) 2014-present Raymond Hill
+
+    This program is free software: you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation, either version 3 of the License, or
+    (at your option) any later version.
+
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with this program.  If not, see {http://www.gnu.org/licenses/}.
+
+    Home: https://github.com/gorhill/uBlock
+*/
+
+/* globals requestIdleCallback, cancelIdleCallback, performance */
+
+'use strict';
+
+/******************************************************************************/
+
+const timer = {};
+
+timer.requestIdleCallback = typeof requestIdleCallback !== 'undefined' ?  requestIdleCallback : null;
+timer.cancelIdleCallback = typeof cancelIdleCallback !== 'undefined' ?  cancelIdleCallback : null;
+
+/******************************************************************************/
+
+const needsPolyfill = timer.requestIdleCallback === null;
+
+if ( needsPolyfill ) {
+    // https://github.com/behnammodi/polyfill/blob/1a5965edc0e2eaf8e6d87902cc719462e2a889fb/window.polyfill.js
+
+    timer.requestIdleCallback = function (callback, options) {
+        options = options || {};
```

Comment:
@gorhill I'm not sure that all of this is useful but I've left it as it is. You had a much simpler polyfill. Let me know if you'd like to use that one.